### ISSUE TITLE
Slow change detection with many files

### ISSUE THREAD

**Issue by AndreyNazarchuk** [nextcloud/client#208]
change detection before each sync is very slow with many files (10,000<)

Onedrive syncs way faster, I think its because they hash the fold and watch for changes, if there's a change they rehash and upload changes and hash, this way, you can start something on one machine and pick up on another, and Onedrive sees there's a different folder hash and check which files to download instead of rechecking everything.

TLDR:: make it not recheck all files when something changes between two sync clients.

**Comment by guruz** [comment]
This is fixed in the _upcoming_ 2.5.0 version of the upstream ownCloud client.

See issue https://github.com/owncloud/client/issues/2297
If you're brave enough to try a not-thoroughly tested build then grab a 2.5.x build from https://download.owncloud.com/desktop/daily/

**Comment by peter-sk** [comment]
This is still happening for me. I'm trying to synch 671 GByte distributed to 436808 files using the Mac desktop app 2.5.1.

My computer is rarely up and running long enough to complete a scan for changes, which takes hours.

**Comment by Eloque** [comment]
Version 2.5.1final (build 20181204)

Still occurring on MacBook Pro, i7 16Gb. 80 GB over few 10.000 files, takes more hours then the notebook is on to check local changes.

**Comment by realies** [comment]
Extremely slow considering the disk speeds and CPU core amount.
